src/LuaAST.ts

```diff
@@ -878,7 +878,7 @@ export function createTableIndexExpression(
     return expression;
 }
 
-export type IdentifierOrTableIndexExpression = Identifier | TableIndexExpression;
+export type LeftHandSideExpression = Identifier | TableIndexExpression;
```

Comment:
I think LeftHandSideExpression is probably a bit _too_ generic. What about `AssignmentLeftHandSideExrpession` or `AssignmentTargetExpression`?